src/progly/query.cc

```diff
@@ -568,7 +646,66 @@ void worker()
+                        int ret ;
+
+                        // extract ptr to meta data blob
+                        const char* blob_dataptr = reinterpret_cast<const char*>( meta.blob_data ) ;
+                        size_t blob_sz           = meta.blob_size ;
+                        //std::cout << "blob_sz = " << blob_sz << std::endl ;
+
+                        // extract bl_seq bufferlist
+                        ceph::bufferlist bl_seq ;
+                        bl_seq.append( blob_dataptr, blob_sz ) ;
+
+                        // bl_seq for ROW format will only contain one bl
+                        ceph::bufferlist::iterator it_bl_seq = bl_seq.begin() ;
+
+                        ceph::bufferlist bl ;
+                        ::decode( bl, it_bl_seq ) ; // this decrements get_remaining by moving iterator
+                        const char* dataptr = bl.c_str() ;
+                        size_t datasz       = bl.length() ;
+                        //std::cout << "datasz = " << datasz << std::endl ;
+
+                        if( meta.blob_format == SFT_FLATBUF_UNION_ROW )
+                            ret = processSkyFb_fbu_rows(
+                                   flatbldr,
+                                   sky_tbl_schema,
+                                   sky_qry_schema,
+                                   sky_qry_preds,
+                                   dataptr,
+                                   datasz,
+                                   errmsg ) ;
+                        else if( meta.blob_format == SFT_FLATBUF_UNION_COL )
+                            ret = processSkyFb_fbu_cols(
+                                    bl_seq,
+                                    flatbldr,
+                                    sky_tbl_schema,
+                                    sky_qry_schema,
+                                    sky_qry_preds,
+                                    dataptr,
+                                    datasz,
+                                    errmsg ) ;
+                        else
+                            assert (Tables::TablesErrCodes::SkyFormatTypeNotRecognized==0);
+
+                        if (ret != 0) {
+                            int more_processing_failure = true;
+                            std::cerr << "ERROR: query.cc: processing flatbuf: "
+                                      << errmsg << "\n Tables::ErrCodes=" << ret
+                                      << endl;
+                            assert(more_processing_failure);
+                        }
+
+                        // TODO: we should be using uint8_t here
+                        const char* processed_data = \
+                            reinterpret_cast<const char*>(flatbldr.GetBufferPointer());
+                        sky_root root = getSkyRoot(processed_data, 0);
+                        result_count += root.nrows;
+                        print_data(processed_data, 0, SFT_FLATBUF_FLEX_ROW);
```

Comment:
i think you are still inside the case SFT_FLATBUF_UNION_ROW/COL block, so the type should be one of the union types instead of SFT_FLATBUF_FLEX_ROW here

Reply:
for expediency, this version of processing fbus packs up the results in the flex rows format, so printing the results will require the flex row format type.

Reply:
oh i see, ok thanks.